kinappserver/models/offer.py

```diff
@@ -131,20 +135,22 @@ def get_cost_and_address(offer_id):
 
 
 def get_offers_for_user(user_id):
-    """return the list of active offers for this user"""
+    """return the list of offers with there status for this user"""
     from distutils.version import LooseVersion
     all_offers = Offer.query.filter_by(is_active=True).order_by(Offer.kin_cost.asc()).all()
     
     # filter out offers with no goods
     redeemable_offers = []
-    from .good import goods_avilable
+    from .good import goods_avilable, not_enought_kin, max_goods_reached
     for offer in all_offers:
-        if goods_avilable(offer.offer_id):
-            redeemable_offers.append(offer)
-        else:
-            #print('filtering out un-redeemable offer-id: %s' % offer.offer_id)
-            pass
-
+        if not goods_avilable(offer.offer_id):
+            offer.unavailable_reason = 'Sold out Check back again soon'
```

Comment:
@katzio we should take into account the client version here - otherwise old clients will see this as an available offer.